src/StarkEx.Crypto.SDK/Signing/StarkExSigner.cs

```diff
@@ -58,6 +59,47 @@ public bool VerifySignature(string messageHash, string publicKey, SignatureModel
             new BigInteger(signature.S.RemoveHexPrefix(), 16));
     }
 
+    /// <inheritdoc />
+    public string GetPrivateStarkKeyFromEthSignature(SignatureModel ethSignature)
+    {
+        var r = ethSignature.R.RemoveHexPrefix();
+
+        if (ethSignature.R.RemoveHexPrefix().Length != 64)
```

Comment:
nitpick: do this validation before line 65

Reply:
Why? R should have exactly 32 bytes, which should translate into a 64 chars hex string, without the `0x` hex prefix.

![image](https://user-images.githubusercontent.com/104689221/224074131-acb39b5e-c20e-4848-b24a-a95de79d1982.png)

https://medium.com/mycrypto/the-magic-of-digital-signatures-on-ethereum-98fe184dc9c7